src/codegen/compile_classical.ml

```diff
@@ -8322,21 +8322,25 @@ module MakeSerialization (Strm : Stream) = struct
         ReadBuf.read_leb128 env get_main_typs_buf ^^ set_arg_count ^^
 
         G.concat_map (fun t ->
-          let can_recover, default_or_trap = Type.(
+          let can_recover, argument_default_or_trap, coercion_default_or_trap = Type.(
+            let null_result _ = Opt.null_lit env in
             match normalize t with
-            | Prim Null | Opt _ | Any ->
-              (Bool.lit true, fun msg -> Opt.null_lit env)
+            | Prim Null ->
+              (get_can_recover, null_result, fun _ -> compile_unboxed_const (coercion_error_value env))
```

Comment:
I don't quite grok why this case is different. Can you explain?

Reply:
```
// parsing reserved as null
assert blob "DIDL\00\01\70" == "(null)" : (reserved) "reserved";
```
Just to clarify, it think the comment should reading `// parsing reserved as null value of type reserved`, if that helps any.